### ISSUE TITLE
Cannot stream chat completions from Azure

### ISSUE THREAD

**Issue by johanbaath** [openai/openai-node#1015]
### Confirm this is a Node library issue and not an underlying OpenAI API issue

- [X] This is an issue with the Node library

### Describe the bug

When streaming chat completions using `client.chat.completions.create` with `AzureOpenAI` client and reading with `ChatCompletionStreamingRunner.fromReadableStream` on the client, the following error occurs:

```
OpenAIError: Cannot read properties of undefined (reading 'content')
```

**Cause:**

The error seems to be caused by `choice.delta` being undefined during the streaming process.

**Questions:**

1. Is this a known issue?
2. Will this library support streaming from Azure, given potential differences in response structure?

### To Reproduce

1. Initialize an  `AzureOpenAI` client
2. Using `client.chat.completions.create` stream a response to the client (a web app)
3. Stream the response using `ChatCompletionStreamingRunner.fromReadableStream` on the client
4. Observe the error

### OS

macOS

### Node version

Node v20.14.0

### Library version

openai 4.56.0

**Comment by deyaaeldeen** [comment]
Hi @johanbaath, could you share a small repro code?

**Comment by johanbaath** [comment]
@deyaaeldeen sure something like this:

Server:
```ts
const azureOpenaiClient = new AzureOpenAI({
  endpoint: "...",
  deployment: "...",
  apiVersion: "2024-07-01-preview",
  apiKey: "...",
});

response = await azureOpenaiClient.chat.completions.create({
  model: "",
  messages: [
    {
      role: "user",
      content: "hello!",
    },
  ],
  stream: true,
});
```

Client:
```ts
const response = await ky.post(endpoint, {
  body,
  signal: abortController.signal,
});

const runner = ChatCompletionStream.fromReadableStream(response.body);

runner.on("content", (delta) => {
  finalText += delta;
});

await runner.finalChatCompletion();
```

**Comment by deyaaeldeen** [comment]
Thanks! Could you also share the name and version of the deployed model and the region it is deployed in?

**Comment by johanbaath** [comment]
@deyaaeldeen gpt-4o, version: 2024-05-13, region: Sweden Central

**Comment by deyaaeldeen** [comment]
@johanbaath Thanks for confirming! Is [asynchronous filter](https://learn.microsoft.com/azure/ai-services/openai/concepts/content-filter?tabs=warning%2Cuser-prompt%2Cpython-new#asynchronous-filter) enabled by any chance? This feature has been known to cause such behavior and a similar issue has been reported elsewhere, for example in https://github.com/openai/openai-python/issues/1677 with more context.

**Comment by johanbaath** [comment]
> @johanbaath Thanks for confirming! Is [asynchronous filter](https://learn.microsoft.com/azure/ai-services/openai/concepts/content-filter?tabs=warning%2Cuser-prompt%2Cpython-new#asynchronous-filter) enabled by any chance? This feature has been known to cause such behavior and a similar issue has been reported elsewhere, for example in [openai/openai-python#1677](https://github.com/openai/openai-python/issues/1677) with more context.

Yes! I use the async filter, is there anything I can do or is this being worked on? Thank you!

**Comment by deyaaeldeen** [comment]
We're discussing this behavior internally and I'll post an update as soon as I have one. For now, I suggest handling the absence of the `delta` property even if the type doesn't say it is possible.

**Comment by johanbaath** [comment]
> We're discussing this behavior internally and I'll post an update as soon as I have one. For now, I suggest handling the absence of the `delta` property even if the type doesn't say it is possible.

Considering that this library does not properly manage this scenario, and given that Azure OpenAI for TypeScript advises migration to openai-node (as detailed [here](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/openai/openai/MIGRATION.md)), are you suggesting that we manually patch the existing library as a temporary solution? Thanks!

**Comment by doubianimehdi** [comment]
@deyaaeldeen this solution doesn't seem to work , this is really blocking , my company is really stuck due to this ... it works with any other GUI ... this is a shame :(

**Comment by deyaaeldeen** [comment]
@doubianimehdi I am sorry that you're experiencing issues in your code. Could you try this complete sample and let us know what issues are you facing exactly?: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/openai/openai/samples/v2/typescript/src/streamChatCompletions.ts

**Comment by Swarnalathaa** [comment]
@deyaaeldeen following is the error response we are getting

![Image](https://github.com/user-attachments/assets/ed0f8131-49b1-4cf4-b84c-0f6d685b54f4)

this is the code block we have
        for await (const chunk of stream) {
          if (chunk?.choices?.[0]?.delta?.reasoning_content) {
            const reasoning_content = chunk?.choices?.[0]?.delta?.reasoning_content || '';
          }
          const delta = chunk?.choices?.[0]?.delta;
          if (delta) {
            const token = chunk?.choices?.[0]?.delta?.content || '';
            if (token) {
              reasoningCompleted = true;
              ......
            }
               ......
            }
  
          }
        }

**Comment by doubianimehdi** [comment]
@deyaaeldeen @Swarnalathaa is working with me , so she answered about your request :)

**Comment by deyaaeldeen** [comment]
@doubianimehdi, @Swarnalathaa thank you for sharing. You need to make sure `delta` is defined before accessing anything underneath it and in the code you provided, there is only one access to `content` and it is guarded by optional chaining operator: `const token = chunk?.choices?.[0]?.delta?.content || '';` which seems ok to me.

**Comment by HAYDEN-OAI** [comment]
thanks for the detailed async filter reproduction! this will be out in the next release via #1982. we updated the streaming runner to tolerate Azure annotation-only chunks that omit `delta`, and to keep empty-ID filter chunks attached to the current completion instead of treating them as a new response. we also added a regression test for the proxied `fromReadableStream()` flow.
